src/amo/components/Categories.js

```diff
@@ -1,29 +1,51 @@
 import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
 import { compose } from 'redux';
 
 import Link from 'amo/components/Link';
-import { visibleAddonType } from 'core/utils';
+import { categoriesFetch } from 'core/actions/categories';
+import { apiAddonType, visibleAddonType } from 'core/utils';
 import translate from 'core/i18n/translate';
+import LoadingText from 'ui/components/LoadingText';
 
 import './Categories.scss';
 
 
 export class CategoriesBase extends React.Component {
   static propTypes = {
     addonType: PropTypes.string.isRequired,
+    dispatch: PropTypes.func,
     categories: PropTypes.object.isRequired,
+    clientApp: PropTypes.string.isRequired,
     error: PropTypes.bool,
     loading: PropTypes.bool.isRequired,
     i18n: PropTypes.object.isRequired,
   }
 
+  componentWillMount() {
+    const { addonType, categories, clientApp, dispatch } = this.props;
+    if (!Object.values(categories).length) {
+      dispatch(categoriesFetch({ addonType, clientApp }));
```

Comment:
Ah thanks, good catch. It works when testing in the browser too (I spent a lot of time looking at it when doing the loading text placeholder) so I knew it worked and didn't give it much thought. 👍